tools/install-powershell.ps1

```diff
@@ -61,16 +76,38 @@ $IsLinuxEnv = (Get-Variable -Name "IsLinux" -ErrorAction Ignore) -and $IsLinux
 $IsMacOSEnv = (Get-Variable -Name "IsMacOS" -ErrorAction Ignore) -and $IsMacOS
 $IsWinEnv = !$IsLinuxEnv -and !$IsMacOSEnv
 
+if ($version) {
+    # if version doesnt contains v prefix, add it
+    if ($version -notmatch "^v") {
+        $version = "v$version"
+    }
+
+    Write-Verbose "Checking if release version '$version' is available" -Verbose
+    $releaseVersions = Invoke-RestMethod https://api.github.com/repos/PowerShell/PowerShell/releases -Verbose:$false
```

Comment:
Is this rate limited?

If so, there should be a switch to bypass this check.

Reply:
Couldn't see a rate limit response code in the [List Releases API Docs](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#list-releases)

Reply:
I could code around this api call if either the file used for `$metadata` was either the metadata.json file in the tools folder and included the latest daily version or the one updated in the `https://aka.ms/pwsh-buildinfo-daily'` file included all versions

I felt that for now this was a best effort update without changing too many other parts made sense for now.

Reply:
Since it is new functionality, I agree that we can address it if it becomes a problem.  We might also, want to write a warning that only the latest version of each release is actually supported.  Therefore, you should use a parameterset that results in us reading metadata.json rather than pinning it to a version.